src/in-memory-backend.service.ts

```diff
@@ -533,9 +534,9 @@ export class InMemoryBackendService {
   }
 
   protected getLocation(href: string) {
-    const l = document.createElement('a');
-    l.href = href;
-    return l;
+    // default the base url when not running in browser
+    let base = document ? document.location.href : 'http://node';
+    return new URL(href, base);
```

Comment:
@steveschmitt I'm worried that this won't work in all the browsers we support because the `URL` type is  [experimental](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL). I'm not sure it's shimmed.

Would you please check in IE, Edge, and FF.  Or come up with an alternative for producing the string we need.  I don't think our use of this location stuff needs to be that complicated. I think we can do it with regEx for example.